### PR TITLE
Finally a decent implementation (alternatively, but less braggy, "todos and fixmes")

### PR DESCRIPTION
**Pull request by darosior**
So i took a shot at fixing the FIXMEs i've been stacking for awhile now. We now probably have more FIXMEs than previously but we're in a *much* better state: i could finally test this randomly-generated text that i write and call 'code'.

This mainly contains:
- **A lot** of bug fixes (dca273c, 6c72af5, 2aac66b, 04d17c8, 03c3c46, 5d86d36, 3c1d249, 1fe6e69)
- Some slight improvements that are going to make our lifes easier (28a109f, b68443e, 8ad1700, 6090307, 17b4c0f)
- Quite some more testing coverage

### CODE REVIEW

**Comment by darosior** [comment]
Fixed windows (again)

File: doc/API.md [hunk 1]
```diff
@@ -31,12 +31,25 @@ Display general information about the current daemon state.
...
+
+### `getdepositaddress`
+
+Get an address to build a deposit transaction.
+
+#### Response
+
+| Field         | Type   | Description                                                 |
+| ------------- | ------ | ----------------------------------------------------------- |
+| `address`     | string | An address for the N-of-N multisig deposit script           |
```
Comment (edouardparis) [inline doc/API.md:43]: Provide the address format

File: src/daemon/bitcoind/interface.rs [hunk 2]
```diff
@@ -432,45 +468,43 @@ impl BitcoinD {
...
-            new_utxos.insert(
+            new_deposits.insert(
                 outpoint,
                 CachedVault {
                     txo: TxOut {
                         value,
                         script_pubkey,
                     },
-                    status: VaultStatus::Funded,
+                    status: if confirmations < 6 {
```
Comment (edouardparis) [inline src/daemon/bitcoind/interface.rs:498]: 6 is reused line 440: time to make it a constant

Comment (darosior) [reply src/daemon/bitcoind/interface.rs:498]: Right, good catch

File: doc/API.md (hunk 1 above)
Comment (darosior) [reply doc/API.md:43]: We only ever support Segwit v0, don't you think that's redundant ?

Comment (darosior) [reply doc/API.md:43]: Will add a commit specifying this for the whole doc

Comment (darosior) [reply doc/API.md:43]: Done

File: src/daemon/bitcoind/interface.rs (hunk 2 above)
Comment (darosior) [reply src/daemon/bitcoind/interface.rs:498]: Done

File: tests/test_misc.py [hunk 3]
```diff
@@ -33,8 +33,37 @@ def test_getinfo(revaultd_manager, bitcoind):
...
+
+    # Generate 5 blocks, it is still unconfirmed
+    bitcoind.generate_block(5)
+    revaultd_manager.rpc.call("listvaults")["vaults"][0]["status"] == \
+        "unconfirmed"
+
+    # 1 more block will get it confirmed
+    bitcoind.generate_block(1)
+    revaultd_manager.rpc.call("listvaults")["vaults"][0]["status"] == \
+        "funded"
```
Comment (darosior) [inline tests/test_misc.py:58]: ffs.. It got me going for half an hour... A nice brainfart over here

Comment (darosior) [reply tests/test_misc.py]: Fixed in 432b8a0

**Comment by darosior** [comment]
Added some more fixup commits, but won't anymore as i don't wanna overload this branch. It's again ready for review

**Comment by darosior** [comment]
Rebased on master with the quicker functional tests..

File: src/daemon/jsonrpc/api.rs [hunk 4]
```diff
@@ -103,41 +101,60 @@ impl RpcApi for RpcImpl {
...
+                        })
+                        .collect::<jsonrpc_core::Result<Vec<VaultStatus>>>()?,
+                )
+            } else {
+                None
+            }
         } else {
             None
         };
         let txids = if let Some(txids) = txids {
```
Comment (edouardparis) [inline src/daemon/jsonrpc/api.rs:130]: Ithink that if you unwrap an option to return an option, use map
```
 let txids = if let Some(txids) = txids {...} else {None};
```
to 
```
let txids = txids.map(|txids|{...});
```

**Review by edouardparis — COMMENTED** [review]
looks good to me, I can ACK if you dont care of the nit

File: src/daemon/jsonrpc/api.rs (hunk 4 above)
Comment (darosior) [reply src/daemon/jsonrpc/api.rs:130]: You can't return an error in a closure.. 
```rust
.collect::<jsonrpc_core::Result<Vec<VaultStatus>>>()?,
```
https://github.com/re-vault/revaultd/pull/46/files/881fa57932918a3046c1afb71d438c0f53cbad55#diff-5416085ff6b3b242b02325c6140edced6cff30147ef1b8d1fc2c30b6b0a23c2bR122

**Review by edouardparis — APPROVED** [review]
ACK 881fa57932918a3046c1afb71d438c0f53cbad55

**Pull request merged**